src/category_theory/limits/over.lean

```diff
@@ -84,6 +86,119 @@ instance forget_preserves_colimits [has_colimits.{v} C] {X : C} :
+/-- Given pullbacks in C, we have pullbacks in C/B -/
+instance {B : C} [has_pullbacks.{v} C] : has_pullbacks.{v} (over B) :=
+begin
+  refine ⟨⟨λ F, _⟩⟩,
+  let X : over B := F.obj walking_cospan.one,
+  let Y : over B := F.obj walking_cospan.left,
+  let Z : over B := F.obj walking_cospan.right,
+  let f : Y ⟶ X := (F.map walking_cospan.hom.inl),
+  let g : Z ⟶ X := (F.map walking_cospan.hom.inr),
+  let L : over B := over.mk (pullback.fst ≫ Y.hom : pullback f.left g.left ⟶ B),
+  let π₁ : L ⟶ Y := over.hom_mk pullback.fst,
+  let π₂ : L ⟶ Z, refine @over.hom_mk _ _ _ L Z (pullback.snd : L.left ⟶ Z.left) _,
+    simp,
+    rw [← over.w f, ← category.assoc, pullback.condition, category.assoc,  over.w g],
+  refine {cone := cone.of_pullback_cone (pullback_cone.mk π₁ π₂ _), is_limit := {lift := _, fac' := _, uniq' := _}},
+    ext, simp, erw pullback.condition,
+  intro s,
+  apply over.hom_mk _ _,
+  apply pullback.lift (s.π.app walking_cospan.left).left (s.π.app walking_cospan.right).left,
+  rw ← over.comp_left, rw ← over.comp_left,
+  rw s.w, rw s.w, simp,
+  show pullback.lift (((s.π).app walking_cospan.left).left) (((s.π).app walking_cospan.right).left) _ ≫
+    (pullback.fst ≫ Y.hom : pullback f.left g.left ⟶ B) = (s.X).hom, simp, refl,
+  intros s j, simp, ext1, dsimp,
+  cases j, simp, simp, simp,
+  show _ ≫ (((pullback_cone.mk π₁ π₂ _).π).app walking_cospan.one).left = ((s.π).app walking_cospan.one).left,
+  dunfold pullback_cone.mk, dsimp,
+  show pullback.lift (((s.π).app walking_cospan.left).left) (((s.π).app walking_cospan.right).left) _ ≫
+    pullback.fst ≫ f.left =
+  ((s.π).app walking_cospan.one).left, simp, rw ← over.comp_left, rw ← s.w walking_cospan.hom.inl,
+  intros s m J, apply over.over_morphism.ext, simp, apply pullback.hom_ext,
+  simp at J, dsimp at J,
+  have := J walking_cospan.left, dsimp at this, simp, rw ← this, simp,
+  have := J walking_cospan.right, dsimp at this, simp, rw ← this, simp
+end
+
+/-- Given pullbacks in C, we have binary products in any over category -/
+instance over_has_prods_of_pullback [has_pullbacks.{v} C] (B : C) :
+  has_binary_products.{v} (over B) :=
+{has_limits_of_shape := {has_limit := λ F, over_product_of_pullbacks B F}}
+
+/-! A collection of lemmas to decompose products in the over category -/
+@[simp] lemma over_prod_pair_left [has_pullbacks.{v} C] {B : C} (f g : over B) :
+  (f ⨯ g).left = pullback f.hom g.hom := rfl
+
+@[simp] lemma over_prod_pair_hom [has_pullbacks.{v} C] {B : C} (f g : over B) :
+  (f ⨯ g).hom = pullback.fst ≫ f.hom := rfl
+
+@[simp] lemma over_prod_fst_left [has_pullbacks.{v} C] {B : C} (f g : over B) :
+  (limits.prod.fst : f ⨯ g ⟶ f).left = pullback.fst := rfl
+
+@[simp] lemma over_prod_snd_left [has_pullbacks.{v} C] {B : C} (f g : over B) :
+  (limits.prod.snd : f ⨯ g ⟶ g).left = pullback.snd := rfl
+
+@[simp] lemma over_prod_map_left [has_pullbacks.{v} C] {B : C} (f g h k : over B) (α : f ⟶ g) (β : h ⟶ k) :
```

Comment:
This shouldn't be `simp`. The left-hand side is somewhat readable. The RHS looks scary to me. Not `simp`ler at all.

Reply:
Alright, I've removed the `simp` attribute here.